src/fu-engine.c

```diff
@@ -594,7 +596,7 @@ fu_engine_modify_remote (FuEngine *self,
 	g_autoptr(GKeyFile) keyfile = g_key_file_new ();
 
 	/* check remote is valid */
-	remote = fu_config_get_remote_by_id (self->config, remote_id);
+	remote = fu_remotes_get_by_id (self->remotes, remote_id);
 	if (remote == NULL) {
 		g_set_error (error,
 			     FWUPD_ERROR,
```

Comment:
So this is still going to rely upon the `inotify` for the remote change event.  How about calling `fu_remotes_reload` directly in this function and maybe destroying/recreating the `GFileMonitor`?

Reply:
I think we can just create a `fu_remote_list_set_key_value()` and make it simpler still. Fixed.